### DISCUSSION TITLE
Referencing Parameter Store secret in add-on template

### DISCUSSION THREAD

**Discussion by gidsg** [aws/copilot-cli#5633] [Q&A] [answered]
Is it possible to reference the value of a Parameter Store secret from within an add-on template? If so how can this be achieved?

(The documentation covers how to do this for a service manifest: https://aws.github.io/copilot-cli/docs/developing/secrets/ but not for an add-on template.)

**Comment by KollaAdithya** [comment]
Hello @gidsg !

Can you explain more about your use case on why want to reference secret from Parameter store in the addon template?

**Reply by gidsg** [reply]
Apologies for late reply, use case here is we want to pass a secret value to an add-on and wanted to see if it could be done using Parameter Store secure string to keep the existing pattern used by copilot.

**Comment by KollaAdithya** [comment, marked as answer]
Hello @gidsg !

Can you take look at this doc https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html
on using dynamic references to specify template values. This will allow you to pass secrets from Parameter Store directly to addon template.
